pallets/msa/src/benchmarking.rs

```diff
@@ -930,6 +930,200 @@ mod benchmarks {
 		Ok(())
 	}
 
+	#[benchmark]
+	fn propose_to_update_provider() -> Result<(), BenchmarkError> {
+		let s = T::MaxProviderNameSize::get();
+		let lang_size = T::MaxLanguageCodeSize::get();
+		let _cid_size = T::MaxLogoCidSize::get();
+		let max_locale_count = T::MaxLocaleCount::get();
+
+		let provider_name = (1..s as u8).collect::<Vec<_>>();
+		let cid = "bafkreidgvpkjawlxz6sffxzwgooowe5yt7i6wsyg236mfoks77nywkptdq"
+			.as_bytes()
+			.to_vec();
+		let mut localized_names = BoundedBTreeMap::new();
+		let mut localized_cids = BoundedBTreeMap::new();
+		for i in 0..max_locale_count {
+			let lang_code = make_lang_code(i as usize, lang_size as usize);
+			let lang = BoundedVec::try_from(lang_code).unwrap();
+			let name = BoundedVec::try_from(provider_name.clone()).unwrap_or_default();
+			let logo = BoundedVec::try_from(cid.clone()).unwrap();
+			localized_names.try_insert(lang.clone(), name).unwrap();
+			localized_cids.try_insert(lang, logo).unwrap();
+		}
+		let account = create_account::<T>("account", 0);
+		let (_, provider_public_key) = Msa::<T>::create_account(account, EMPTY_FUNCTION).unwrap();
+		let entry = ProviderRegistryEntry {
+			default_name: BoundedVec::try_from(provider_name).unwrap_or_default(),
+			localized_names,
+			default_logo_250_100_png_cid: BoundedVec::try_from(cid).unwrap(),
+			localized_logo_250_100_png_cids: localized_cids,
+		};
+
+		// Must be an already registered provider to propose update
+		assert_ok!(Msa::<T>::create_provider_v2(
+			RawOrigin::Signed(provider_public_key.clone()).into(),
+			ProviderRegistryEntry::default()
+		));
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(provider_public_key), entry);
+
+		assert_eq!(frame_system::Pallet::<T>::events().len(), 1);
+		Ok(())
+	}
+
+	#[benchmark]
+	fn update_application_via_governance() -> Result<(), BenchmarkError> {
```

Comment:
I think we probably want this benchmark to be parameterized based on the number of locales included in the update, no?